src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ControllerExtensions.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Tenants.ViewModels;
+
+namespace OrchardCore.Tenants.Controllers
+{
+    public static class ControllerExtensions
+    {
+        public static async Task ValidateModelAsync<TViewModel>(this Controller controller, TViewModel model, bool newTenant = true) where TViewModel : TenantViewModel
```

Comment:
Looking at this method, it actually just looks like it's all validation.

It would probably suit `IValidatableObject` better, than being an extension method or service.

Reply:
It could I will make the last change, coz my aim to make it service like what you suggested in another PR

Reply:
If the controller implement `IValidatableObject` the issue still same, the code will be duplicate in both controller